### PR TITLE
test: verify file url interception works as expected

### LINKED ISSUE #1506: page content incorrectly embedded when using file protocol
Hi again,
As usual, thanks a lot for maintaining puppeteer - it's super useful. 👍 

Seems to me that there was a regression in `0.13` 🐛  when using the combination of:
- `file:///` protocol
- `setRequestInterception: true`

### Steps to reproduce

**Tell us about your environment:**

* Puppeteer version: `0.13.0`
* Platform / OS version: `MacOS High Sierra`, same on `windows`
* URLs (if applicable):
Any url over the `file` protocol seems to have this problem.

**What steps will reproduce the problem?**

_Please include code that reproduces the issue._
```js
const puppeteer = require('puppeteer');
(async() => {
const browser = await puppeteer.launch({
      // waitUntil: 'networkidle2', // see if this performs better than default for timeouts
      ignoreHTTPSErrors: true,
      args: ['--disable-setuid-sandbox', '--no-sandbox']
    })
const page = await browser.newPage()

// CRITICAL: _only_ with setRequestInterception = true do we get this problem:
// the real html is escaped and embedded in a template like below.
// <html><head></head><body><pre style="word-wrap: break-word; white-space: pre-wrap;">
// <REAL CONTENT HERE>
// </pre></body></html>
await page.setRequestInterception(true)
page.on('request', interceptedRequest => {
  console.log(interceptedRequest.url)
  interceptedRequest.continue()
})

// ANY local html file will do
await page.goto('file:///Users/pocketjoso/penthouse/test/static-server/yeoman.html')

const content = await page.content()
console.log('content: ' + content)

browser.close();
})();

```
**What is the expected result?**
The html is parsed in the same way regardless of file/http protocol, and regardless of whether `setRequestInterception` is true or false. 

**What happens instead?**
```
content: <html><head></head><body><pre style="word-wrap: break-word; white-space: pre-wrap;">&lt;!doctype html&gt;
&lt;html class="no-js"&gt;
    &lt;head&gt;
        &lt;meta charset="utf-8"&gt;
        &lt;title&gt;critical css test&lt;/title&gt;
        &lt;meta name="description" content=""&gt;
        &lt;meta name="viewport" content="width=device-width, initial-scale=1"&gt;
        &lt;!-- Place favicon.ico and apple-touch-icon.png in the root directory --&gt;
... etc
```
Or in a picture (the browser renders the markup as text, rather than as html):
bjor6laez-after

### PR DESCRIPTION
**Pull request by aslushnikov**
This was fixed upstream: https://crrev.com/550319
Rolled into pptr: https://github.com/GoogleChrome/puppeteer/pull/2393

Fixes #1506.

### CODE REVIEW

File: test/network.spec.js [hunk 1]
```diff
@@ -459,6 +459,18 @@ module.exports.addTests = function({testRunner, expect}) {
...
       expect(error.message).toContain('Request Interception is not enabled');
     });
+    it('should work with file URLs', async({page, server}) => {
+      await page.setRequestInterception(true);
+      const urls = new Set();
+      page.on('request', request => {
+        urls.add(request.url().split('/').pop());
+        request.continue();
+      });
+      await page.goto('file://' + path.join(__dirname, 'assets', 'one-style.html'));
```
Comment (JoelEinbinder) [inline test/network.spec.js]: Path.join will have the wrong way slashes on windows

Comment (aslushnikov) [reply test/network.spec.js]: I wonder how did it pass on appveyor.

Comment (JoelEinbinder) [reply test/network.spec.js]: I think Chrome auto corrects slashes. A bit iffy to rely on that though.

Comment (aslushnikov) [reply test/network.spec.js]: @JoelEinbinder agreed. Translated the path to URL on our end.

**Review by JoelEinbinder — APPROVED** [review]

**Pull request merged**